java/src/main/java/com/powsybl/python/network/NetworkModificationCFunctions.java

```diff
@@ -0,0 +1,108 @@
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.ConnectablePosition;
+import com.powsybl.python.commons.CTypeUtil;
+import com.powsybl.python.commons.Directives;
+import com.powsybl.python.commons.PyPowsyblApiHeader;
+import org.graalvm.nativeimage.IsolateThread;
+import org.graalvm.nativeimage.ObjectHandle;
+import org.graalvm.nativeimage.ObjectHandles;
+import org.graalvm.nativeimage.c.CContext;
+import org.graalvm.nativeimage.c.function.CEntryPoint;
+import org.graalvm.nativeimage.c.type.CCharPointer;
+
+import static com.powsybl.dataframe.network.adders.SeriesUtils.applyBooleanIfPresent;
+import static com.powsybl.dataframe.network.adders.SeriesUtils.applyIfPresent;
+import static com.powsybl.python.commons.Util.doCatch;
+import static com.powsybl.python.network.NetworkCFunctions.createDataframe;
+
+/**
+ * @author Coline Piloquet <coline.piloquet at rte-france.com>
+ */
+@CContext(Directives.class)
+public final class NetworkModificationCFunctions {
+
+    private NetworkModificationCFunctions() {
+    }
+
+    @CEntryPoint(name = "createFeederBay")
+    public static void createFeederBay(IsolateThread thread, ObjectHandle networkHandle,
+                                       CCharPointer busbarSectionId, int injectionPositionOrder,
+                                       CCharPointer injectionDirection, boolean throwException,
+                                       ObjectHandle reporterHandle, PyPowsyblApiHeader.DataframePointer cDataframeLine,
+                                       PyPowsyblApiHeader.ElementType elementType,
+                                       PyPowsyblApiHeader.ExceptionHandlerPointer exceptionHandlerPtr) {
+
+        doCatch(exceptionHandlerPtr, () -> {
+            String busbarSectionIdStr = CTypeUtil.toString(busbarSectionId);
+            String directionStr = CTypeUtil.toString(injectionDirection);
+            ConnectablePosition.Direction direction = ConnectablePosition.Direction.valueOf(directionStr);
+            Network network = ObjectHandles.getGlobal().get(networkHandle);
+            UpdatingDataframe df = createDataframe(cDataframeLine);
+            InjectionAdder<?> adder = createAdder(elementType, network.getBusbarSection(busbarSectionIdStr).getTerminal().getVoltageLevel(), df);
+            ReporterModel reporter = ObjectHandles.getGlobal().get(reporterHandle);
+            CreateFeederBay modification = new CreateFeederBay(adder, busbarSectionIdStr, injectionPositionOrder, direction);
+            if (reporter == null) {
+                modification.apply(network, throwException, Reporter.NO_OP);
+            } else {
+                modification.apply(network, throwException, reporter);
+            }
+        });
+    }
+
+    public static InjectionAdder<?> createAdder(PyPowsyblApiHeader.ElementType elementType, VoltageLevel voltageLevel, UpdatingDataframe df) {
+        if (elementType == PyPowsyblApiHeader.ElementType.LOAD) {
```

Comment:
I am really not sure about this piece of code. The code is already present in the different DataframeAdder create method, but contains the call to the add() function that I don't need. Maybe it would be possible to enhance that.

Reply:
yes I think it would be better to add a method to create the adder in the DataframeAdder and the create method will just call these one and call create()